### PR TITLE
Implement smart-wallet and read-only UI

### LINKED ISSUE #5356: Smart wallets can get balances with zero transactions (for small # of users)
Epic milestone to https://github.com/Agoric/agoric-sdk/issues/3995

## Acceptance criteria

- When using the wallet, Keplr is not brought up until there is a transfer of value
- UI can show wallet balances
- Subscriptions publish to Cosmos events

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #5356

## Description

Here's a (rather unimpressive) working demonstration of the smart wallet UI.  Notice the zero-latency initialization of the view, and that it's currently read-only.

https://user-images.githubusercontent.com/457244/173690436-b93d6014-5ab7-4f53-966d-aa9fc72dc5e1.mov

### CODE REVIEW

File: packages/vats/src/core/basic-behaviors.js [hunk 1]
```diff
@@ -224,10 +224,41 @@ export const makeAddressNameHubs = async ({
...
+  const marshaller = E(board).getPublishingMarshaller();
   return E(client).assignBundle([
-    address => ({ bank: E(bankManager).getBankForAddress(address) }),
+    address => {
+      const bank = E(bankManager).getBankForAddress(address);
+      const myAddressNameAdmin = E(namesByAddressAdmin).lookupAdmin(address);
+      const myWallet = E(zoe).startInstance(
+        smartWallet,
+        {},
+        { agoricNames, bank, namesByAddress, myAddressNameAdmin, board },
```
Comment (dckc) [inline packages/vats/src/core/basic-behaviors.js:256]: Both the `smartWallet` contract an `ag-solo` use `myAddressNameAdmin` to set up `depositFacet`. There can be only one. This is why we see 0 invitations in xnet https://github.com/Agoric/agoric-sdk/issues/5513#issuecomment-1156612543 .

**Review by dckc — CHANGES_REQUESTED** [review]
found a critical problem

File: packages/wallet/contract/src/smart-wallet.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ *
+ * TODO: multi-tennant wallet
+ *
+ * @param {ZCF} zcf
+ * @param {{
+ *   storageNode?: ERef<StorageNode>,
+ *   marshaller?: ERef<Marshaller>,
+ * }} privateArgs
+ */
+export const start = async (zcf, privateArgs) => {
```
Comment (turadg) [inline packages/wallet/contract/src/smart-wallet.js:26]: we need some test of this contract starting and the shape of its `creatorFacet`. I can help with that if you like.

File: packages/wallet/api/src/lib-wallet.js [hunk 3]
```diff
@@ -582,6 +581,7 @@ export function makeWallet({
 
   const getAttBrand = makeMemoizedGetter(() => E(getAttIssuer()).getBrand());
 
+  /** @typedef {any} AttestationTool */ /* @@@ TODO */
```
Comment (turadg) [inline packages/wallet/api/src/lib-wallet.js:584]: before landing or after?

**Review by turadg — CHANGES_REQUESTED** [review]

**Comment by dckc** [comment]
overtaken by #5701 etc.

**Pull request closed without merging**